nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
 
         # If all else fails, return home. Ideally this should never happen.
         return reverse("home")
+
+
+class NautobotViewSetMixin:
```

Comment:
Why are we not deriving anything from `rest_framework.viewsets.ViewSetMixin`? Wouldn't that be a better base than starting our own new implementation?